app/scripts/controllers/network/network.js

```diff
@@ -194,21 +193,27 @@ export default class NetworkController extends EventEmitter {
       this.emit(NETWORK_EVENTS.INFURA_IS_UNBLOCKED);
     }
 
-    ethQuery.sendAsync({ method: 'net_version' }, (err, networkVersion) => {
-      const currentNetwork = this.getNetworkState();
-      if (initialNetwork === currentNetwork) {
-        if (err) {
-          this._setNetworkState('loading');
-          // keep network details in sync with network state
-          this._clearNetworkDetails();
-          return;
-        }
-
-        this._setNetworkState(networkVersion);
-        // look up EIP-1559 support
-        this.getEIP1559Compatibility();
+    let networkVersion;
+    try {
+      networkVersion = await this._getNetworkId();
+    } catch (error) {
+      if (initialNetwork !== this.getNetworkState()) {
+        return;
       }
-    });
+
+      this._setNetworkState('loading');
+      // keep network details in sync with network state
+      this._clearNetworkDetails();
+      return;
+    }
+
+    if (initialNetwork !== this.getNetworkState()) {
+      return;
+    }
+
+    this._setNetworkState(networkVersion);
+    // look up EIP-1559 support
+    await this.getEIP1559Compatibility();
```

Comment:
Did this suggestion make sense? I believe this would be more straightforward to read:

``` suggestion
    let networkVersion;
    let error;
    try {
      networkVersion = await getNetworkVersion(ethQuery);
    } catch (e) {
      console.error(e);
      error = e;
    }
    if (initialNetwork === this.getNetworkState()) {
      if (error) {
        this.setNetworkState('loading');
        // keep network details in sync with network state
        this.clearNetworkDetails();
        return;
      }
      this.setNetworkState(networkVersion);
      // look up EIP-1559 support
      await this.getEIP1559Compatibility();
    }
```

Reply:
OK, I'll try something along these lines

Reply:
Attempt made here: https://github.com/MetaMask/metamask-extension/pull/16881/commits/1d57a4e90816b00b4db09d687ce9ed58e5ab8cd7

I kept the early-exit to avoid unnecessary indentation, and used `else` over `return` to simplify the control flow. And I didn't add any logging, as I'm trying to avoid additional functional changes. But it does use the error capture strategy you used here.

Reply:
Okay no problem. That looks better. Thanks!